lib/main.js

```diff
@@ -128,81 +129,91 @@ ava.test.before(async () => {
-        ],
-        assert: [ 'The device should successfully get provisioned and appear in dashboard' ]
+      tap.test(`${options.deviceType}: Provision quad model`, async (test) => {
+        return utils.runManualTestCase(test, {
+          do: [
+            'Go into an existing ts4900 app or create a new one',
+            'Select "quad" as "CPU Cores"',
+            'Select any "Network Connection" option',
+            'Download the image and boot a single core variant of TS4900'
+          ],
+          assert: [ 'The device should successfully get provisioned and appear in dashboard' ]
+        })
       })
-    })
-  }
-}
-
-for (const testCase of [
-  require('../tests/bluetooth-test'),
-  require('../tests/device-online'),
-  require('../tests/enter-container'),
-  require('../tests/hdmi-uart5'),
-  require('../tests/hostapp-update'),
-  require('../tests/identification-led'),
-  require('../tests/kernel-splash-screen'),
-  require('../tests/os-file-format'),
-  require('../tests/push-container'),
-  require('../tests/push-multicontainer'),
-  require('../tests/reboot-with-app'),
-  require('../tests/reload-supervisor'),
-  require('../tests/resin-device-progress'),
-  require('../tests/resin-splash-screen'),
-  require('../tests/resin-sync'),
-  require('../tests/rpi-serial-uart0'),
-  require('../tests/rpi-serial-uart1'),
-  require('../tests/service-variables')
-]) {
-  if (testCase.interactive && !options.interactiveTests) {
-    continue
-  }
-
-  if (testCase.deviceType) {
-    const ajv = new AJV()
-    if (!ajv.compile(testCase.deviceType)(deviceTypeContract)) {
-      continue
     }
   }
 
-  ava.test(_.template(testCase.title)({
-    options
-  }), async (test) => {
-    await testCase.run(test, context, options, {
-      resinio
+  Bluebird
+    .try(() => {
+      for (const testCase of [
+        require('../tests/bluetooth-test'),
+        require('../tests/device-online'),
+        require('../tests/enter-container'),
+        require('../tests/hdmi-uart5'),
+        require('../tests/hostapp-update'),
+        require('../tests/identification-led'),
+        require('../tests/kernel-splash-screen'),
+        require('../tests/os-file-format'),
+        require('../tests/push-container'),
+        require('../tests/push-multicontainer'),
+        require('../tests/reboot-with-app'),
+        require('../tests/reload-supervisor'),
+        require('../tests/resin-device-progress'),
+        require('../tests/resin-splash-screen'),
+        require('../tests/resin-sync'),
+        require('../tests/rpi-serial-uart0'),
+        require('../tests/rpi-serial-uart1'),
+        require('../tests/service-variables')
+      ]) {
+        if (testCase.interactive && !options.interactiveTests) {
+          continue
+        }
+
+        if (testCase.deviceType) {
+          const ajv = new AJV()
+          if (!ajv.compile(testCase.deviceType)(deviceTypeContract)) {
```

Comment:
Can we use https://github.com/resin-io-modules/skhema, our new JSON Schema module, and replace AJV?

Reply:
We should, but maybe a different PR